src/hypercube.rs

```diff
@@ -0,0 +1,66 @@
+//! This module defines basic types related to Boolean hypercubes.
```

Comment:
Since Nova does not require this functionality and is needed only for the multi-folding scheme, does it make sense to move this to the ccs module?